mbed_lstools/lstools_base.py

```diff
@@ -73,6 +73,8 @@ def __init__(self):
         self.lock_file = os.path.join(MbedLsToolsBase.HOME_DIR,
                                       MbedLsToolsBase.MBEDLS_HOME_DIR, MbedLsToolsBase.MBEDLS_GLOBAL_LOCK)
         self.mbedls_get_mocks()
+        # Make sure retargeting is applied if retarget file exists
+        self.retarget(True)
```

Comment:
When you're using the API directly, this implementation has the downside of not being able to disable retargetting.

I'd suggest instead of passing in a parameter to retarget, add an optional parameter to the constructor of this class that enables retarget by default. Something like `def __init__(self, retarget=True):` Then you can just conditionally call `retarget` based on this variable.